ADR/0011-roles-and-permissions.md

```diff
@@ -0,0 +1,81 @@
+
+Date: 2023-01-10
+
+## Status
+
+Accepted
+
+## Context
+StoneSoup is using Kubernetes as the control plane for managing its resources. We require a system for managing user roles and permissions in this context. We have defined the following roles for our project: "Contributor", "Maintainer", and "Owner". We need to map these roles to specific permissions in the Kubernetes RBAC system, in terms of API groups, verbs, and resources.
+
+## Decision
+
+We will use the built-in Kubernetes RBAC system for StoneSoup's role and permissions management, and map the following roles to specific permissions, as described in the table below:
+
+### Roles
+Contributor: Members who interact with the workspace mostly through pull requests.
+Maintainer: Members who manage the workspace without access to sensitive or destructive actions.
+Owner: Members who have full access to the workspace including sensitive and destruction actions.
+
+### Roles and Permissions Table
+
+|     Role      | Permissions             | API Groups                | Verbs                             | Resources
+|---------------|-------------------------|---------------------------|-----------------------------------|----------------------------------------------------------------------
+| Contributor   | Workspace               | Access to namespaces that backs workspace                     |
+|               | Application & Component | appstudio.redhat.com      | get, list, watch                  | applications, components, componentdetectionqueries
+|               | Environment             | appstudio.redhat.com      | get, list, watch                  | promotionruns, snapshotenvironmentbindings, snapshots, environments
+|               | *GitOps*                | managed-gitops.redhat.com | get, list, watch                  | gitopsdeployments
+|               | PipelineRun             | tekton.dev                | create, get, list, watch          | pipelineruns
+|               | Pipeline Results        | results.tekton.dev        | get, list                         | results, records
+|               | IntegrationTestScenario | appstudio.redhat.com      | get, list, watch                  | integrationtestscenarios
+|               | Enterprise contract     | appstudio.redhat.com      | get, list, watch                  | enterprisecontractpolicies
+|               | Release Strategy        | appstudio.redhat.com      | get, list, watch                  | releases, releasestrategies, releaseplans
+|               | Release Admission Plan  | appstudio.redhat.com      | get, list, watch                  | releaseplanadmissions
+|               | *JVM Build Service*     | jvmbuildservice.io        | get, list, watch                  | jbsconfigs, artifactbuilds
+|               | *Service Access*        | appstudio.redhat.com      | get, list, watch                  | spiaccesstokenbindings, spiaccesschecks, spiaccesstokens, spiaccesstokendataupdates
```

Comment:
spiaccesstokendataupdates is not used by User (SA only), can be removed
spifilecontentrequest should be added